corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -149,7 +176,7 @@ hqDefine('data_dictionary/js/data_dictionary.js', function () {
         this.toggleGroup = function (group) {
             group.toggle();
             var groupIndex = _.findIndex(self.casePropertyList(), function (element) {
-                return element.name() === group.name();
+                return element.name === group.name;
```

Comment:
Same case type.

> it doesn't make sense to have two with the same name

Yeah, but someone's going to do it, just want to know that the page isn't going to break when they try. Makes sense that groups with the same name would be merged on page reload.